src/main/java/com/karankumar/bookproject/ui/book/form/DoubleToRatingScaleConverter.java

```diff
@@ -35,6 +35,6 @@
 
     @Override
     public Double convertToPresentation(RatingScale rating, ValueContext valueContext) {
-        return RatingScale.toDouble(rating);
+        return RatingScale.toDouble(rating).orElse(0.0);
```

Comment:
In this case, we do want to return `null` in the else clause. Otherwise, this will change the semantics (`null` rating = no rating, which is different from a rating of 0/10).

We can change this to the following:

```java
return RatingScale.toDouble(rating).get();
```

Reply:
This came to my mind when I worked on this part as well. Thanks a lot for clarifying!

Reply:
It seems that just using `get()` here will produce some errors with `BookFormTest.java`. Using `orElse(null)` will get rid of those errors, but I'm not sure if this is the best approach since our goal is to reduce the usage of `null` in the first place.

Reply:
> It seems that just using get() here will produce some errors with BookFormTest.java.

That's strange, it shouldn't. Perhaps you're getting a Flyway error and need to clean? Either way, if you need a hand with this, feel free to ask on the #help channel on Slack and post the stack trace using something like GitHub Gist or Pastebin.

> but I'm not sure if this is the best approach since our goal is to reduce the usage of null in the first place.

Generally speaking, this is true to the extent that we don't introduce breaking changes. `convertToPresentation` is an overridden converter method that takes a `RatingScale` value and converts it into a `Double`.  In our `RatingScale` enum, we've set the `NO_RATING` to have the numeric value `null`. That's different to `ZERO` (a value of `0.0`). 

In other words, `null` is acceptable in this case as we handle this specifically. While we could change `value` to be `Optional`, I think that adds unnecessary noise with no real gain

Reply:
> It seems that just using get() here will produce some errors with BookFormTest.java.

Ignore me, you're right (`return RatingScale.toDouble(rating).orElse(null)`)

To test this, you can try adding a book to the read shelf. If you don't set a rating, it should show `No rating`, not `0/10`. This is why we need the `orElse(null)`